docs/design/seccomp.md

```diff
@@ -0,0 +1,299 @@
+    2.  To be able to grant access to a seccomp profile to a class of users
+2.  As a user, I want to run an application with a seccomp profile similar to
+    the default one provided by my container runtime
+3.  As a user, I want to run an application which is already libseccomp-aware
+    in a container, and for my application to manage interacting with seccomp
+    unmediated by Kubernetes
+4.  As a user, I want to be able to describe a custom seccomp profile and use
+    it with my containers
+
+### Use Case: Administrator access control
+
+Controlling to control access to seccomp profiles is a cluster administrator concern.
+It should be possible for an administrator to:
+
+1.  Define a default seccomp profile for a class of users
+2.  Grant access to a seccomp profile to a class of users
+
+Administrators should be able to control two aspects of defaults for seccomp
+profiles:
+
+1.  Defaults for classes of users based on pod security policy
+2.  Defaults at the API level
+
+#### Classes of users
+
+The [pod security policy]() API extension governs the ability of users to make
+requests that affect pod and container security contexts.  The proposed design
+should deal with required changes to control access to new API features added.
+
+#### API Defaults
+
+There should be a minimum API-level guarantee for the seccomp profile to use
+when the field is not set when a pod is admitted to the API server.  This
+profile should be exposed in the control plane as an argument to the API
+server.
+
+### Use Case: Seccomp profiles similar to container runtime defaults
+
+Many users will want to use images that make assumptions about running in the
+context of their chosen container runtime.  Such images are likely to
+frequently assume that they are running in the context of the container
+runtime's default seccomp settings.  Therefore, it should be possible to
+express a seccomp profile similar to a container runtime's defaults.
+
+### Use Case: Applications that link to libseccomp
+
+Some applications already link to libseccomp and control seccomp directly.  It
+should be possible to run these applications unmodified in Kubernetes; this
+implies there should be a way to disable seccomp control in Kubernetes for
+certain containers, or to run with a "no-op" or "unconfined" profile.
+
+### Use Case: Custom profiles
+
+Different applications have different requirements for seccomp profiles; it
+should be possible to express arbitrary seccomp profiles and use them in
+containers.
+
+## Community Work
+
+### Container runtime support for seccomp
+
+#### opencontainers
+
+Docker and `runc` both support the open container initiative's API for
+seccomp, which is very close to the libseccomp API.  It allows full
+specification of seccomp filters, with arguments, operators, and actions.
+
+#### appcontainers
+
+The `rkt` runtime delegates to systemd for seccomp support.  The `appc`
+project has an [open issue](https://github.com/appc/spec/issues/529) to be
```

Comment:
Yeah and a default profile specifically

On Thu, Apr 21, 2016 at 10:44 AM, Paul Morie <notifications@github.com>
wrote:

> In docs/design/seccomp.md
> <https://github.com/kubernetes/kubernetes/pull/24602#discussion_r60624574>
> :
>
> > +containers.
> > +
> > +## Community Work
> > +
> > +### Container runtime support for seccomp
> > +
> > +#### opencontainers
> > +
> > +Docker and `runc` both support the open container initiative's API for
> > +seccomp, which is very close to the libseccomp API.  It allows full
> > +specification of seccomp filters, with arguments, operators, and actions.
> > +
> > +#### appcontainers
> > +
> > +The `rkt` runtime delegates to systemd for seccomp support.  The `appc`
> > +project has an [open issue](https://github.com/appc/spec/issues/529) to be
>
> Do you mean, adding richer seccomp filtering capabilities?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/24602/files/735c581d4346a5aaddf7de6bd52089c870c3d107#r60624574>
>



-- 


Jessie Frazelle
4096R / D4C4 DD60 0D66 F65A 8EFC  511E 18F3 685C 0022 BFF3
pgp.mit.edu <http://pgp.mit.edu/pks/lookup?op=get&search=0x18F3685C0022BFF3>